src/actions/testActions.ts

```diff
@@ -264,29 +270,26 @@ export const startTests =
             } else {
                 logger.info(`End test failed: ${message}`);
             }
-            dispatch(cleanup());
         });
 
         dispatch(startedAction(testMode));
     };
+
 const cleanup = (): AppThunk<RootState, Promise<void>> => async dispatch => {
-    if (dtm.dtmTransport.port.isOpen) {
-        await dtm.dtmTransport.close();
-        dtm = null;
-    }
+    await disposeDTM();
     dispatch(stoppedAction());
 };
 
 export const endTests = () => {
     logger.info('Ending test');
-    dtm.endTest();
+    getDTM()?.endTest();
 };
 
 export const deselectDevice =
-    (): AppThunk<RootState> => (dispatch, getState) => {
+    (): AppThunk<RootState, Promise<void>> => async (dispatch, getState) => {
         const isRunning = getIsRunning(getState());
         if (isRunning) {
             endTests();
         }
-        dispatch(cleanup());
+        await dispatch(cleanup());
```

Comment:
Why the await. Nothing would get blocked by this anway.
If it takes too long we should handle this in other ways since user can still select another device before this is done (in the case it really does take long)

Reply:
Just for completion... this way if call deselectDevice we can also await if we end up reusing same port after deselect awaiting make sure that we first await the closing of the first  one and then opening new port... 99.9 it will not make any diff but to me it looks more accurate .